src/libasr/codegen/llvm_utils.cpp

```diff
@@ -1245,11 +1245,37 @@ namespace LFortran {
         return LLVM::CreateLoad(*builder, value_ptr);
     }
 
-    llvm::Value* LLVMList::read_item(llvm::Value* list, llvm::Value* pos, bool get_pointer,
+    llvm::Value* LLVMList::get_list_pos(llvm::Module& module, llvm::Value* list, llvm::Value* pos) {
+        llvm::Value* len = LLVMList::len(list);
+        llvm::Value* zero = llvm::ConstantInt::get(llvm::Type::getInt32Ty(context),
+                                                   llvm::APInt(32, 0));
+
+        llvm::Function *fn = builder->GetInsertBlock()->getParent();
+        llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
+        llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
+        llvm::BasicBlock *mergeBB = llvm::BasicBlock::Create(context, "ifcont");
+
+        llvm::Value* cond = builder->CreateICmpSGT(zero, pos);
+        builder->CreateCondBr(cond, thenBB, elseBB);
+        builder->SetInsertPoint(thenBB);
+        {
+            pos = builder->CreateAdd(len, pos);
```

Comment:
This happens because you are storing into a register conditionally. Instead create a temporary variable outside if-else block and store pos in it. Then load temporary variable inside if and else separately, update it and then store back into the temporary variable.